src/datasets/features.py

```diff
@@ -471,15 +473,15 @@ class ClassLabel:
             classes, one per line.
     """
 
-    num_classes: int = None
-    names: List[str] = None
-    names_file: str = None
+    num_classes: Optional[int] = None
+    names: Optional[List[str]] = None
+    names_file: Optional[str] = None
```

Comment:
Currently it's true that these can stay None.
We should disallow that though by doing some changes in the post_init.
This is out of scope for this PR though, so let's keep the Optional for now